src/libYARP_os/src/yarp/os/NetworkClock.cpp

```diff
@@ -111,7 +112,11 @@ bool NetworkClock::Private::read(ConnectionReader& reader)
 
     listMutex.lock();
     auto waiter_it = waiters->begin();
-    while (waiter_it != waiters->end()) {
+    // Update the wake-up time. In case of a time reset it closes the gap between the waiter and _time.
```

Comment:
I think we need to do this only in the case of clock reset?

Reply:
yes, if you do this all the time, you will never wake up...